test/integration_test.go

```diff
@@ -558,15 +561,39 @@ func (ts *IntegrationTestSuite) TestSignalWorkflow() {
 		ts.tracer.GetTrace("SignalWorkflow"))
 }
 
+func (ts *IntegrationTestSuite) TestSignalWorkflowWithInterceptorError() {
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+
+	// Return error 3 times from the interceptor
+	ts.inboundSignalInterceptor.returnErrorTimes = 3
+	wfOpts := ts.startWorkflowOptions("test-signal-workflow-interceptor-error")
+	run, err := ts.client.ExecuteWorkflow(ctx, wfOpts, ts.workflows.SignalWorkflow)
+	ts.Nil(err)
+	// Let workflow task run and send signal after to ensure correct order.
+	<-time.After(time.Second)
+	err = ts.client.SignalWorkflow(ctx, "test-signal-workflow-interceptor-error", run.GetRunID(), "string-signal", "string-value")
+	ts.NoError(err)
+
+	wt := &commonpb.WorkflowType{Name: "workflow-type"}
+	err = ts.client.SignalWorkflow(ctx, "test-signal-workflow-interceptor-error", run.GetRunID(), "proto-signal", wt)
+	ts.NoError(err)
+
+	var protoValue *commonpb.WorkflowType
+	err = run.Get(ctx, &protoValue)
+	// Workflow should succeed after retries upon an error in the signal interceptor
```

Comment:
Ideally we'd somehow verify the "signal is dropped" behavior we want here. It's not super obvious to me if/how this test is doing that

Reply:
Yes, as it stands test verifies that we don't break existing behavior. It's a little tricky as we need to verify that workflow task didn't run, hence can't test it from the workflow itself. I thought about checking metrics but that proven to be flaky.